Source/WebCore/automation/AutomationInstrumentation.h

```diff
@@ -0,0 +1,72 @@
+/*
+* Copyright (C) 2024 Igalia S.L.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are
+* met:
+*
+*     * Redistributions of source code must retain the above copyright
+* notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above
+* copyright notice, this list of conditions and the following disclaimer
+* in the documentation and/or other materials provided with the
+* distribution.
+*     * Neither the name of Google Inc. nor the names of its
+* contributors may be used to endorse or promote products derived from
+* this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+* A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+* OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+* DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+* THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#pragma once
+
+#if ENABLE(WEBDRIVER_BIDI)
+
+#include <JavaScriptCore/ConsoleMessage.h>
+#include <wtf/AbstractRefCountedAndCanMakeWeakPtr.h>
+#include <wtf/CompletionHandler.h>
+#include <wtf/Function.h>
+#include <wtf/MemoryPressureHandler.h>
+#include <wtf/ObjectIdentifier.h>
+#include <wtf/RefPtr.h>
+
+namespace Inspector {
+class ConsoleMessage;
+}
+
+namespace WebCore {
+
+
+class WEBCORE_EXPORT AutomationInstrumentationClient : public AbstractRefCountedAndCanMakeWeakPtr<AutomationInstrumentationClient> {
+public:
+    virtual ~AutomationInstrumentationClient() = default;
+
+    virtual void addMessageToConsole(const Inspector::ConsoleMessage&);
+};
+
+
+class WEBCORE_EXPORT AutomationInstrumentation {
+public:
+    static void setClient(const AutomationInstrumentationClient&);
+    static void clearClient();
+
+    static void addMessageToConsole(const std::unique_ptr<Inspector::ConsoleMessage>&);
+};
+
+inline void AutomationInstrumentationClient::addMessageToConsole(const Inspector::ConsoleMessage&)
+{
+}
```

Comment:
NIT: i'd just inline this above

Reply:
Do you mean moving this right after the `AutomationInstrumentionClient` declaration, before the `AutomationInstrumentation` class?

Reply:
i mean having it be right when you first declare it
```cpp
static void addMessageToConsole(const std::unique_ptr<Inspector::ConsoleMessage>&) { }
```

Reply:
Oh, right. I accidentally had left something inside the header declaration, which the compiler complained about. I'll update it.